Api/WebScrap.API/Extract.cs

```diff
@@ -14,34 +13,34 @@ public static class Extract
     /// <param name="css"></param>
     /// <returns></returns>
     public static ImmutableArray<string> Html(
-        string html,
-        string css)
+        ReadOnlySpan<char> html,
+        ReadOnlySpan<char> css)
     {
         html = html.TrimStart(' ');
-        var tagFactory = new TagFactory();
-        var tagIndexes = CssProcessor.CalculateTagIndexes(tagFactory, html, css);
-        var tagRanges = ExtractTagRanges(tagFactory, html, tagIndexes);
-        var tagStrings = ExtractStrings(html, tagRanges);
+        var tagIndexes = CssAPI.GetTagIndexes(css, html);
+        var tagRanges = ExtractTagRanges(html.ToString(), tagIndexes);
+        var tagStrings = ExtractStrings(html.ToString(), tagRanges);
         return tagStrings;
     }
 
     private static ImmutableArray<Range> ExtractTagRanges(
-        TagFactory tagFactory,
         string html,
         IEnumerable<int> tagIndexes)
         => tagIndexes.Select(tagIndex =>
         {
+            // Deprecated? 
+            // TODO: please remove. The processorBase already returns all of that.
             var substring = html[tagIndex..];
```

Comment:
// Deprecated? // TODO: please remove. The processorBase already returns all of that.